functional/server_test.go

```diff
@@ -0,0 +1,133 @@
+// across reconfiguration of fleetd after receiving SIGHUP.
+func TestReconfigureServer(t *testing.T) {
+	cluster, err := platform.NewNspawnCluster("smoke")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cluster.Destroy()
+
+	m0, err := cluster.CreateMember()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = cluster.WaitForNMachines(m0, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// NOTE: we need to sleep once here to get reliable results.
+	// Without this sleep, the entire fleetd test always ends up succeeding
+	// no matter whether SIGHUP came or not.
+	_, _ = cluster.MemberCommand(m0, "sh", "-c", `'sleep 2'`)
+
+	err = waitForFleetdSocket(cluster, m0)
+	if err != nil {
+		t.Fatalf("Failed to get a list of fleetd sockets: %v", err)
+	}
+
+	// send a SIGHUP to fleetd, and periodically checks if a message
+	// "Reloading configuration" appears in fleet's journal, up to timeout (15) seconds.
+	stdout, _ := cluster.MemberCommand(m0, "sudo", "kill", "-HUP", "$(pidof fleetd)")
+	if strings.TrimSpace(stdout) != "" {
+		t.Fatalf("Sending SIGHUP to fleetd returned: %s", stdout)
+	}
+
+	err = waitForReloadConfig(cluster, m0)
+	if err != nil {
+		t.Fatalf("Failed to get log about reconfiguration: %v", err)
+	}
+
+	// check if fleetd is still running correctly, by running fleetctl status
+	_, _, err = cluster.Fleetctl(m0, "status")
+	if err != nil {
+		t.Fatalf("Unable to check status. Please check if fleetd still holds the correct socket. err:%v", err)
+	}
+
+	// Check for HTTP listener error looking into the fleetd journal
+	stdout, _ = cluster.MemberCommand(m0, "journalctl -u fleet")
+	if strings.Contains(strings.TrimSpace(stdout), "Failed serving HTTP on listener:") {
+		t.Fatalf("Fleetd log returned error on HTTP listeners: %s", stdout)
+	}
+
+	// Check expected state after reconfiguring fleetd
+	stdout, _ = cluster.MemberCommand(m0, "systemctl", "show", "--property=ActiveState", "fleet")
+	if strings.TrimSpace(stdout) != "ActiveState=active" {
+		t.Fatalf("Fleet unit not reported as active: %s", stdout)
+	}
+	stdout, _ = cluster.MemberCommand(m0, "systemctl", "show", "--property=Result", "fleet")
+	if strings.TrimSpace(stdout) != "Result=success" {
+		t.Fatalf("Result for fleet unit not reported as success: %s", stdout)
+	}
+}
+
+// waitForReloadConfig returns if a message "Reloading configuration" exists
+// in the journal, periodically checking for the journal up to the timeout.
+func waitForReloadConfig(cluster platform.Cluster, m0 platform.Member) (err error) {
+	_, err = util.WaitForState(
+		func() bool {
+			// NOTE: journalctl should run just simply like "journalctl -u fleet",
+			// without being piped with grep. Doing
+			// "journalctl -u fleet | grep \"Reloading configuration\"" is racy
+			// in a subtle way, so that it sometimes fails only on semaphoreci.
+			// - dpark 20160408
+			stdout, _ := cluster.MemberCommand(m0, "journalctl -u fleet")
+			journalfleet := strings.TrimSpace(stdout)
```

Comment:
I don't see why `pidof` would be more reliable than `-u`? Do we expect more than one instance of fleetd to exist during the test? (And if so, is it actually appropriate to consider only the currently active one?...)

(I don't see why using grep wouldn't be reliable in the first place, though... But doing the compare in Go is fine too; so I guess this is a non-issue.)